### PR TITLE
Rubrik Solution Update (#2)

### PR DESCRIPTION
**Pull request by benmeadowcroft**
Required items, please complete
   
   Change(s):
   - Updated playbooks, accommodating changes in deprecated APIs on Rubrik Security Cloud
   - Updated documentation and package

   Reason for Change(s):
   - Resolves issues with deprecated API usage

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - TBD

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @benmeadowcroft, please resolve the validations below and repackage the solution. Thanks

image

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyAnalysis/azuredeploy.json [hunk 1]
```diff
@@ -22,8 +23,8 @@
             "6. Repeat steps for other connections"
         ],
         "prerequisitesDeployTemplateFile": "Azure-Sentinel/Solutions/RubrikSecurityCloud/Playbooks/RubrikCustomConnector/Rubrik_custom_conn.json",
-        "lastUpdateTime":  "2022-09-26T00:00:00.000Z",
-        "entities":  ["account","url"],
+        "lastUpdateTime":  "2022-01-20T00:00:00.000Z",
+        "entities":  ["account","url","api"],
```
Comment (v-rbajaj) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyAnalysis/azuredeploy.json:27]: Hi @benmeadowcroft "API" isn't a valid entity, please check the following document --> https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#entity-mapping

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/azuredeploy.json [hunk 2]
```diff
@@ -23,11 +23,9 @@
             "2. Configure the automation rules to trigger the playbook."
         ],
         "prerequisitesDeployTemplateFile": "Azure-Sentinel/Solutions/RubrikSecurityCloud/Playbooks/RubrikCustomConnector/Rubrik_custom_conn.json",
-        "lastUpdateTime": "2022-09-26T00:00:00.000Z",
-        "entities":  [
-        ],
-        "tags":  [
-        ],
+        "lastUpdateTime":  "2022-01-20T00:00:00.000Z",
+        "entities":  ["account","url","api"],
```
Comment (v-rbajaj) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/azuredeploy.json:27]: Hi @benmeadowcroft "API" isn't a valid entity, please check the following document --> https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#entity-mapping

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikRansomwareDiscoveryAndVMRecovery/azuredeploy.json [hunk 3]
```diff
@@ -24,8 +25,8 @@
             "6. Repeat steps for other connections"
         ],
         "prerequisitesDeployTemplateFile": "Azure-Sentinel/Solutions/RubrikSecurityCloud/Playbooks/RubrikCustomConnector/Rubrik_custom_conn.json",
-        "lastUpdateTime": "2022-09-26T00:00:00.000Z",
-        "entities":  ["account","url"],
+        "lastUpdateTime":  "2022-01-20T00:00:00.000Z",
+        "entities":  ["account","url","api"],
```
Comment (v-rbajaj) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikRansomwareDiscoveryAndVMRecovery/azuredeploy.json:29]: Hi @benmeadowcroft "API" isn't a valid entity, please check the following document --> https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#entity-mapping

**Comment by v-rbajaj** [comment]
Hi @manishkumar1991 and @rahul0216 can you please check the playbook changes?

**Review by rahul0216 — COMMENTED** [review]
I'm getting this error while deploying Data Object Discovery Playbook and similar sort of error with IncidentResponse playbook. Looks like playbook need to be deployed in some order or some playbooks are dependent on others. From the readme file, that relation is not clear to me.
Can you put that detail in readme file or if I'm missing something, let me know.
image

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikRansomwareDiscoveryAndVMRecovery/azuredeploy.json (hunk 3 above)
Comment (benmeadowcroft) [reply Solutions/RubrikSecurityCloud/Playbooks/RubrikRansomwareDiscoveryAndVMRecovery/azuredeploy.json]: Thanks, the team should have resolved this in [a839e7f](https://github.com/Azure/Azure-Sentinel/pull/7352/commits/a839e7fbbfe41ed7fbd38f589e9397c1f38a5243)

**Comment by benmeadowcroft** [comment]
Hi @rahul0216 , have updated the readme in [2557aa8](https://github.com/Azure/Azure-Sentinel/pull/7352/commits/2557aa82a929cb5875e44472ebd822efc3a00470) to address your comment

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikDataObjectDiscovery/azuredeploy.json [hunk 4]
```diff
@@ -3,11 +3,12 @@
...
         "prerequisites":  [
             "1. The Rubrik Security Cloud data connector should be configured to send appropriate events to Microsoft Sentinel.",
             "2. The Rubrik Security Cloud solution should be configured to [connect to Rubrik Security Cloud API end points using a Service Account](https://docs.rubrik.com/en-us/saas/saas/polaris_api_access_with_service_accounts.html), the service account should be assigned a role that includes the relevant privileges necessary to perform the desired operations (see [Roles and Permissions](https://docs.rubrik.com/en-us/saas/saas/common/roles_and_permissions.html) in the Rubrik Security Cloud user guide).",
-            "3. Store Service account credentials in Key Vault and obtain keyvault name and tenantId",
+            "3. Rubrik custom connector needs to be deployed prior to the deployment of this playbook, in the same resource group and region. Relevant instructions can be found in the connector doc page.",
+            "4. Store Service account credentials in Key Vault and obtain keyvault name and tenantId",
                 "a. Create a Key Vault with unique name",
```
Comment (rahul0216) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikDataObjectDiscovery/azuredeploy.json:12]: Add the prerequisite playbook details here as well. Also, I don't see corresponding readme file have been updated with the prerequisite playbook details.

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/README.md [hunk 5]
```diff
@@ -9,6 +9,12 @@ This playbook provides an end to end example of the collection of Ransomware Ano
 1. The Rubrik Security Cloud data connector should be configured to send appropriate events to Microsoft Sentinel.
 2. The Rubrik Security Cloud solution should be configured to [connect to Rubrik Security Cloud API end points using a Service Account](https://docs.rubrik.com/en-us/saas/saas/polaris_api_access_with_service_accounts.html), the service account should be assigned a role that includes the relevant privileges necessary to perform the desired operations (see [Roles and Permissions](https://docs.rubrik.com/en-us/saas/saas/common/roles_and_permissions.html) in the Rubrik Security Cloud user guide).
 3. Obtain Teams group id and channel id.
+4. Make sure that below mentioned playbooks are deployed before deploying RubrikAnomalyIncidentResponse playbook:
+    * RubrikDataObjectDiscovery
+    * RubrikAnomalyAnalysis
+    * RubrikFilesetRansomwareDiscovery
+    * RubrikRansomwareDiscoveryAndFileRecovery
+    * RubrikRansomwareDiscoveryAndVMRecovery
```
Comment (rahul0216) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/README.md:18]: Correct the spelling 'paramteres'.

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/azuredeploy.json [hunk 6]
```diff
@@ -23,11 +23,9 @@
             "2. Configure the automation rules to trigger the playbook."
         ],
```
Comment (rahul0216) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/azuredeploy.json:24]: Add playbook prerequisite step here as well.

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikFilesetRansomwareDiscovery/azuredeploy.json [hunk 7]
```diff
@@ -19,7 +19,8 @@
             "5. Click Save",
             "6. Repeat steps for other connections"
         ],
```
Comment (rahul0216) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikFilesetRansomwareDiscovery/azuredeploy.json:21]: Add the playbook prerequisite step here as well.

**Comment by v-atulyadav** [comment]
Hi @benmeadowcroft, please check @rahul0216 comments. Thanks.

**Comment by benmeadowcroft** [comment]
> Hi @benmeadowcroft, please check @rahul0216 comments. Thanks.

@v-atulyadav yes, the feedback is being worked on by the development team, will update the PR when I get the fixes from them

**Comment by git-rubrik** [comment]
Submitted change to fix requests from @rahul0216 and @v-atulyadav in [5baed4a](https://github.com/Azure/Azure-Sentinel/pull/7352/commits/5baed4aec2f9ade92568367da27a17ff39ca6e90)

**Review by rahul0216 — APPROVED** [review]
Thanks. Changes look good.

**Comment by v-atulyadav** [comment]
Will check on this validation failures. Thanks

**Comment by v-atulyadav** [comment]
Hi @benmeadowcroft, we have fixed armttk validations in the attached MainTemplate.json file. Please use this file in your zip and PR. Also request you to upgrade the solution version from 2.0.0 to 2.0.1
[mainTemplate.zip](https://github.com/Azure/Azure-Sentinel/files/10920703/mainTemplate.zip)

**Review by rahul0216 — APPROVED** [review]
Changes look good.

File: Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/azuredeploy.json [hunk 8]
```diff
@@ -7,7 +7,13 @@
...
             "1. The Rubrik Security Cloud data connector should be configured to send appropriate events to Microsoft Sentinel.",
             "2. The Rubrik Security Cloud solution should be configured to [connect to Rubrik Security Cloud API end points using a Service Account](https://docs.rubrik.com/en-us/saas/saas/polaris_api_access_with_service_accounts.html), the service account should be assigned a role that includes the relevant privileges necessary to perform the desired operations (see [Roles and Permissions](https://docs.rubrik.com/en-us/saas/saas/common/roles_and_permissions.html) in the Rubrik Security Cloud user guide).",
-            "3. Obtain Teams group id and channel id."
+            "3. Obtain Teams group id and channel id.",
+            "4. Make sure that below mentioned playbooks are deployed before deploying RubrikAnomalyIncidentResponse playbook:
+                - RubrikDataObjectDiscovery
+                - RubrikAnomalyAnalysis
+                - RubrikFilesetRansomwareDiscovery
+                - RubrikRansomwareDiscoveryAndFileRecovery
+                - RubrikRansomwareDiscoveryAndVMRecovery"
```
Comment (rahul0216) [inline Solutions/RubrikSecurityCloud/Playbooks/RubrikAnomalyIncidentResponse/azuredeploy.json:16]: This string (point 4) is not formatted correctly, causing the ARM template failure.

**Review by rahul0216 — CHANGES_REQUESTED** [review]
@benmeadowcroft  Please address the comment. It is causing the validation failure.

**Comment by v-atulyadav** [comment]
Hi @benmeadowcroft, we have fixed armttk validations suggested by Rahul in the attached playbook file. Please use this file and repackage the solution.
[azuredeploy.zip](https://github.com/Azure/Azure-Sentinel/files/10964595/azuredeploy.zip)

**Comment by git-rubrik** [comment]
Added edits based on @rahul0216 suggestions.

**Review by rahul0216 — DISMISSED** [review]
Changes look good. Thanks

**Comment by v-atulyadav** [comment]
Hi @benmeadowcroft @git-rubrik, It looks like there is a mismatch between MainTemplate.json inside the zip file and MainTemplate.json outside the zip file.

image

And also in data connector ispreview set to true instead of false.
image

 To maintain consistency in both files, please repackage the solution.

**Comment by benmeadowcroft** [comment]
> Hi @benmeadowcroft @git-rubrik, …
> To maintain consistency in both files, please repackage the solution.

Hi @v-atulyadav, it looks like the wrong zip was merged into the PR. I have attempted to provide the correctly packaged zip now. Please let me know if this looks okay to you now? Thank you for your assistance.

**Comment by v-atulyadav** [comment]
Hi @benmeadowcroft, thanks for your response will check and respond you accordingly.

**Comment by v-atulyadav** [comment]
> And also in data connector ispreview set to true instead of false. image
> 
Hi @benmeadowcroft, please reply on this why ispreview set to true instead of false?

Please include the old version of zip 2.0.0 as well, as you have removed it from PR
image

**Comment by benmeadowcroft** [comment]
> Hi @benmeadowcroft, please reply on this why ispreview set to true instead of false?

I have committed a change to switch this value with this release
 
> Please include the old version of zip 2.0.0 as well, as you have removed it from PR

Okay, I have added back the 2.0.0.zip file. Should we continue to keep these previous zip packages in the repository as we make changes in the future?

**Comment by v-atulyadav** [comment]
> Okay, I have added back the 2.0.0.zip file. Should we continue to keep these previous zip packages in the repository as we make changes in the future?

Yes, we should keep previous zip packages.

**Comment by benmeadowcroft** [comment]
@v-atulyadav thanks for the confirmation. I think this has addressed all the open comments at this point.

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**